src/v2/context.in.rs

```diff
@@ -32,6 +32,21 @@ impl Context {
             Context::Dir(Path::new(&s_ref).to_owned())
         }
     }
+
+    /// Determines if two Contexts are equivalent. This is only true
+    /// if they are the same directory path, or if they are same git
+    /// repository and branch. Subdirectories of a repository may differ
+    pub fn is_equivalent_to(&self, other: &Context) -> bool {
```

Comment:
Yeah I totally agree. Really, the whole purpose of this function is to figure out whether two sources conflict, i.e. [this check](https://github.com/faradayio/cage/blob/master/src/sources.rs#L75). So it's very driven by Cage's requirements, which is to know whether or not two contexts with the same alias are compatible.

Given that it's so Cage-specific, I do wonder if it belongs in `cage/src/ext/context.rs`? Then it could have a more revealing name, like `conflicts_with`? The parsing of the git URL could stay in `compose_yml`, as that is a bit more general-purpose.

Reply:
Failing that, a better name might be `is_compatible_with`? It feels a little bit better, though you'd probably still need to read its docs to know what exactly "compatible" means.